From 6ed312fb09d455d5c6fbe5114f07e9a76ff9605a Mon Sep 17 00:00:00 2001
From: Chris Diamand <chris@diamand.org>
Date: Wed, 1 Apr 2015 17:52:07 +0100
Subject: [PATCH 3/4] Provide a non-inlined version of __is_aU().

This means the Clang sanitiser doesn't have to recreate the entire function
(with caching, etc) in hand-written LLVM IR.
---
 src/libcrunch.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/src/libcrunch.c b/src/libcrunch.c
index 6b2391d..8bcb8ad 100644
--- a/src/libcrunch.c
+++ b/src/libcrunch.c
@@ -21,6 +21,7 @@
 #endif
 #include "libcrunch.h"
 #include "libcrunch_private.h"
+#include "libcrunch_cil_inlines.h"
 
 #define NAME_FOR_UNIQTYPE(u) ((u) ? ((u)->name ?: "(unnamed type)") : "(unknown type)")
 
@@ -1611,3 +1612,11 @@ can_hold_pointer_failed:
 	return 1; // fail, but program continues
 	
 }
+
+/* Provide a non-inlined version of __is_aU(). This means the Clang sanitiser
+ * doesn't have to recreate the entire function (including caching) in
+ * hand-written LLVM IR. */
+int __is_aU_not_inlined(const void *obj, const void *uniqtype)
+{
+	__is_aU(obj, uniqtype);
+}
-- 
2.1.4

